oak-commons/src/test/java/org/apache/jackrabbit/oak/commons/LongUtilsTest.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jackrabbit.oak.commons;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+public class LongUtilsTest {
+
+    @Test
+    public void tryParse() {
+        assertEquals(-1, LongUtils.tryParse("-1").longValue());
```

Comment:
We need tests for just "-", "-0", and specially for empty string.

Reply:
I would add tests cases for ranges "MIN_VALUE .. MIN_VALUE + 100" and "MAX_VALUE - 100 .. MAX_VALUE".

But specially, I would add a lot more tests cases for out-of-range numbers.

Reply:
Done. Please let me know if there are more tests that should be added. I also changed the tests to compare with `java.util.Long#parseLong`. That is the logic that we are replacing, so the results should be the same.